metaflow_extensions/plugins/project_environment.py

```diff
@@ -73,3 +178,28 @@ def add_to_package(self):
     def get_client_info(cls, *args):
         """Client information."""
         return "Local project environment (metaflow_extensions)"
+
+    def bootstrap_commands(self, step_name):
+        """Run before any step decorators are initialized."""
+        # Import encapsulated to avoid import errors
+        from metaflow_extensions.config.metaflow_config import PREINSTALL_PKGS
+
+        # Run any pre-install scripts
+        flow_name = Path(sys.argv[0]).stem
+        preinstalls = (
+            "preinstall.sh",
+            f"preinstall-{flow_name}.sh",
+            f"preinstall-{flow_name}-{step_name}.sh",
+        )
+        preinstalls = list(filter(lambda fname: Path(fname).exists(), preinstalls))
```

Comment:
Just come across an error, need to incorporate the flow directory into the path resolution - the current working directory may not be the same as the flow directory....

```suggestion
        flow_name = Path(sys.argv[0]).stem
        flow_dir = Path(sys.argv[0]).parent
        preinstalls = (
            "preinstall.sh",
            f"preinstall-{flow_name}.sh",
            f"preinstall-{flow_name}-{step_name}.sh",
        )
        preinstalls = list(
            filter(lambda fname: (flow_dir / fname).exists(), preinstalls)
        )
```